### ISSUE TITLE
Support for Bastion / Jumphosts?

### ISSUE THREAD

**Issue by prologic** [tenstad/terraform-provider-remote#98]
Can we add support for Bastion / Jumphosts?

The only thing that needs to happen is:

- Add an optional `bastion { ... }` block
- If bastion is defined, first dial a connection to the bastion host, get an ssh client back, then dial the actual target host.

**Comment by tenstad** [comment]
Thank you for the suggestion! I'm not familiar with bastions. What would the `bastion` block need to contain? The same configuration as the current `conn` block?

**Comment by prologic** [comment]
Yes more or less. The same connection details like user/password + host or host + key are used to make an initial connection to a host, then with the resulting connection you make the actual connection to the host you're targeting.

The SSH library implements the Dialer and Conn interfaces.

**Comment by jorikseldeslachts** [comment]
Any progress on this feature?

**Comment by tenstad** [comment]
Not that I am aware of. Would you be interested in looking into it and creating a PR?

**Comment by alita1991** [comment]
Hi, does anyone have any update/workaround to handle the jump host?
